proxylib/proxylib/connection.go

```diff
@@ -30,9 +30,7 @@ import (
 // caller and passed in again when more data has been received on the
 // connection.
 
-type Direction struct {
-	InjectBuf *[]byte
-}
+type InjectBuf *[]byte
```

Comment:
Added this:
```
// InjectBuf is a pointer to a slice header for an inject buffer allocated by
// the proxylib caller. As data is placed into the buffer, the length
// of the buffer in the slice header is increased correspondingly. To make
// the the injected data visible to the caller we need to pass the slice header
// by reference rather than by value, hence the pointer in the type.
// As the caller is typically in a differnent memory management domain (not
// subject to Go runtime garbage collection), the underlying buffer may never
// be expanded or otherwise reallocated.
```

Reply:
I used the same opportunity to comment the Connection struct and its fields.